java/src/test/java/ai/rapids/cudf/TableTest.java

```diff
@@ -7291,4 +7291,21 @@ void testExplodeOuterPosition() {
       }
     }
   }
+
+  @Test
+  void testSample() {
+    Table t = new Table.TestBuilder().column("s1", "s2", "s3", "s4", "s5").build();
```

Comment:
This leaks `t`, `ret` (three times), and each of the scalar elements in the first sample.

There are also cleaner ways to compare columns as well.

```java
try (Table t = new Table.TestBuilder().column("s1", "s2", "s3", "s4", "s5").build()) {
  try (Table ret = t.sample(3, false, 0);
         Table expected = new Table.TestBuilder().column().build("s3", "s4", "s5")) {
    assertTablesAreEqual(expected, ret);
  }

  try (Table ret = t.sample(100, true, 0)) {
    assertEquals(100, ret.rowCount());
  }

  try (Table ret = t.sample(4, true, 0)) {
    assertEqual(4, ret.getRowCount());
  }
}
```